### PR TITLE
More requests/items service refactoring

### PR DESCRIPTION
**Pull request by kenoir**
We can further reduce what is available in the shared "stacks" library as most of the code now splits neatly into items or requests.

For https://github.com/wellcomecollection/catalogue-api/pull/241

### CODE REVIEW

File: requests/src/test/scala/weco/api/requests/services/SierraRequestsServiceTest.scala
Comment (kenoir) [inline requests/src/test/scala/weco/api/requests/services/SierraRequestsServiceTest.scala:33]: These tests are replaced by the extra case here: https://github.com/wellcomecollection/catalogue-api/pull/242/files#diff-9ed61b263ed01f0a064581dfb2ce9ab7ecdaa54dc15a08b555c08fbf1b6e6cb7R239

**Review by alexwlchan — COMMENTED** [review]
The SierraSource fixture has also been added in scala-libs in a separate PR – is it worth waiting until that's merged and used here before reviewing?

**Comment by kenoir** [comment]
> The SierraSource fixture has also been added in scala-libs in a separate PR – is it worth waiting until that's merged and used here before reviewing?

I'll come back and do that in a different PR. I think this change is substantial enough.

File: items/src/main/scala/weco/api/items/services/SierraItemUpdater.scala
Comment (alexwlchan) [inline items/src/main/scala/weco/api/items/services/SierraItemUpdater.scala:77]: Probably not for fixing here, but it occurs to me that the Sierra API does give us a list of bib IDs in the item responses, and we could just pick one to use here.

File: items/src/main/scala/weco/api/items/services/SierraItemUpdater.scala
Comment (alexwlchan) [inline items/src/main/scala/weco/api/items/services/SierraItemUpdater.scala:105]: nit

```suggestion
          warn(s"Item lookup missing items: $missingItems")
```

File: items/src/main/scala/weco/api/items/services/SierraItemUpdater.scala
Comment (alexwlchan) [inline items/src/main/scala/weco/api/items/services/SierraItemUpdater.scala:108]: nit

```suggestion
          error(s"Item lookup failed: $itemLookupError")
```

File: items/src/main/scala/weco/api/items/services/SierraItemUpdater.scala
Comment (alexwlchan) [inline items/src/main/scala/weco/api/items/services/SierraItemUpdater.scala:127]:
```suggestion
      // has not yet propagated. In that case it gets method "NotRequestable")
```

I would add here that this should be extremely rare in practice – items that are removed are usually suppressed in Sierra, then deleted after a long delay. (Do we handle suppressed items correctly?)

We should log a warning here so we can see if it's happening a lot.

File: items/src/main/scala/weco/api/items/services/SierraItemUpdater.scala
Comment (alexwlchan) [inline items/src/main/scala/weco/api/items/services/SierraItemUpdater.scala:139]:
```suggestion
          _ -> AccessCondition(method = AccessMethod.NotRequestable)
```

**Review by alexwlchan — APPROVED** [review]

File: items/src/main/scala/weco/api/items/services/SierraItemUpdater.scala
Comment (kenoir) [reply items/src/main/scala/weco/api/items/services/SierraItemUpdater.scala:105]: I think it's time to add this config to `scalafmt` 🙂 !
```
rewrite.rules = [SortImports, RedundantBraces]
rewrite.redundantBraces.stringInterpolation = true
```

File: items/src/main/scala/weco/api/items/services/SierraItemUpdater.scala
Comment (kenoir) [reply items/src/main/scala/weco/api/items/services/SierraItemUpdater.scala:77]: Oh, that is useful.

**Pull request merged**
